packages/kit/types/index.d.ts

```diff
@@ -1804,6 +1804,11 @@ declare module '@sveltejs/kit' {
 	 * @param e The object to check.
 	 * */
 	export function isRedirect(e: unknown): e is Redirect_1;
+	/**
+	 * Checks whether this is an action failure thrown by {@link fail}.
+	 * @param e The object to check.
+	 * */
+	export function isActionFailure(e: unknown): e is ActionFailure_1<undefined>;
```

Comment:
Through this I'm realizing just now that we're not returning "is this interface" but rather "is this class" from that. We should definitely change it for the new function to refer to the interface instead, not sure if it's seen as a breaking change to change it for the others, too

Reply:
Do you mean we should change them to not use `instanceof` for the check?

Reply:
If there's a breaking change we want to eventually make for the others, can you file an issue and add it to the 3.0 milestone so that we don't forget?

Reply:
I've created one here https://github.com/sveltejs/kit/issues/12986